main/src/ca/uwaterloo/flix/language/phase/TypeReconstruction.scala

```diff
@@ -417,10 +417,13 @@ object TypeReconstruction {
       val eff = Type.mkUnion(eff1 :: es.map(_.eff), loc)
       TypedAst.Expr.Do(op, es, tpe, eff, loc)
 
-    case KindedAst.Expr.InvokeMethod2(exp, name, exps, tvar, evar, loc) =>
+    case KindedAst.Expr.InvokeMethod2(exp, name, exps, mvar, tvar, evar, loc) =>
       val e = visitExp(exp)
-      val es = exps.map(visitExp(_))
-      TypedAst.Expr.InvokeMethod2(e, name, es, subst(tvar), subst(evar), loc)
+      val es = exps.map(visitExp)
+      // Translation from invokeMethod2 to invokeMethod
+      // TODO INTEROP check this
+      val Type.Cst(TypeConstructor.JvmMethod(method), loc) = subst(mvar)
+      TypedAst.Expr.InvokeMethod(method, e, es, subst(tvar), subst(evar), loc)
```

Comment:
The above is correct, but if the pattern match fails we get a Scala MatchError.

We prefer to use InternalCompilerException, so make the pattern match explicit and add a default case that throws that exception.